sdk/eventgrid/eventgrid/test/public/eventGridClientV2.spec.ts

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { Suite, Context } from "mocha";
+import { assert } from "@azure/test-utils";
+import { Recorder, env } from "@azure-tools/test-recorder";
+import {
+  EventGridClient,
+  CloudEvent,
+  ApiManagementGatewayCreatedEventData,
+  ReceiveResult,
+  RejectResult,
+} from "../../src";
+import { createRecordedV2Client } from "./utils/recordedClient";
+import { expect } from "chai";
+
+/* eslint no-constant-condition: "off" */
+describe("Event Grid Client V2", function (this: Suite) {
+  let recorder: Recorder;
+  let client: EventGridClient;
+  let eventSubscripionName: string;
+  let topicName: string;
+  let maxDeliveryCount: number;
+
+  beforeEach(async function (this: Context) {
+    ({ client, recorder } = await createRecordedV2Client(
+      this.currentTest,
+      "EVENT_GRID_V2_ENDPOINT",
+      "EVENT_GRID_V2_KEY"
+    ));
+
+    eventSubscripionName = env["EVENT_SUBSCRIPTION_NAME"] ?? "testsubscription1";
+    topicName = env["TOPIC_NAME"] ?? "testtopic1";
+    maxDeliveryCount = env["MAX_DELIVERY_COUNT"] ? parseInt(env["MAX_DELIVERY_COUNT"]) : 5;
+
+    // Clear any messages that may be available in the topic.
+    while (true) {
+      const receivedResult: ReceiveResult<ApiManagementGatewayCreatedEventData> =
+        await client.receiveCloudEvents(topicName, eventSubscripionName);
+      if (!receivedResult || receivedResult.value.length === 0) break;
+      const lockToken = receivedResult.value[0].brokerProperties.lockToken;
+      await client.acknowledgeCloudEvents([lockToken], topicName, eventSubscripionName);
+    }
```

Comment:
one way to avoid the constant condition:
```suggestion
   let receivedResult: ReceiveResult<ApiManagementGatewayCreatedEventData> =
        await client.receiveCloudEvents(topicName, eventSubscripionName);
    while (receivedResult && receivedResult.value.length > 0) {
       const lockToken = receivedResult.value[0].brokerProperties.lockToken;
       await client.acknowledgeCloudEvents([lockToken], topicName, eventSubscripionName);
       receivedResult = await client.receiveCloudEvents(topicName, eventSubscripionName);
    }
```